examples/cholesky_test.plv

```diff
@@ -0,0 +1,209 @@
+import cholesky;
+
+static diagonal {n}
+  (d :: double[n])
+  (out m :: double[n, n])
+  :: Void := (
+  for i in 0:n ->
+    for j in 0:n ->
+      if i == j then (m[i, j] <- d[i]) else (m[i, j] <- 0.0);
+);
+
+static tril {m, n}
+  (a :: double[m, n])
+  (out b :: double[m, n])
+  :: Void := (
+  for i in 0:m ->
+    for j in 0:n ->
+      if i < j then b[i, j] <- 0.0 else b[i, j] <- a[i, j];
+);
+
+static triu {m, n}
+  (a :: double[m, n])
+  (out b :: double[m, n])
+  :: Void := (
+  for i in 0:m ->
+    for j in 0:n ->
+      if i > j then b[i, j] <- 0.0 else b[i, j] <- a[i, j];
+);
```

Comment:
(sorry the built in upper triangular matrix type requires square dimensions). one alternative expression is:
`vec i in 0:m -> vec j in 0:n -> if i > j then 0.0 else a[i,j];`
then write
`b <- triu a`